store/tikv/gcworker/gc_worker.go

```diff
@@ -560,28 +735,30 @@ func doGC(ctx goctx.Context, store tikv.Storage, safePoint uint64, identifier st
 		}
 
 		if err != nil {
+			log.Warnf("[gc worker] gc for range [%v, %v) safepoint: %v, failed, err: %v", startKey, endKey, safePoint, err)
+			lastErr = err
 			failedRegions++
-			gcActionRegionResultCounter.WithLabelValues("fail").Inc()
-			log.Warnf("[gc worker] %s failed to do gc on region(%s, %s), ignore it", identifier, string(loc.StartKey), string(loc.EndKey))
 		} else {
 			successRegions++
-			gcActionRegionResultCounter.WithLabelValues("success").Inc()
 		}
 
 		key = loc.EndKey
-		if len(key) == 0 {
+		if len(key) == 0 || bytes.Compare(key, endKey) >= 0 {
 			break
 		}
-		bo = tikv.NewBackoffer(tikv.GcOneRegionMaxBackoff, ctx)
 	}
-	log.Infof("[gc worker] %s finish gc, safePoint: %v, successful regions: %v, failed regions: %v, cost time: %s",
-		identifier, safePoint, successRegions, failedRegions, time.Since(startTime))
-	gcHistogram.WithLabelValues("do_gc").Observe(time.Since(startTime).Seconds())
-	return nil
+
+	res := &gcResult{
+		successRegions: successRegions,
+		failedRegions:  failedRegions,
+		lastErr:        lastErr,
+	}
+
+	return res, nil
 }
 
 // these two errors should not return together, for more, see the func 'doGC'
```

Comment:
This comment is strange, would you update it please?

Reply:
It should be here, not outdated, we sadly rely on this.